administrator/components/com_media/resources/scripts/components/browser/items/item.js

```diff
@@ -37,14 +37,63 @@ export default {
             return store.state.selectedItems.some(selected => selected.path === item.path);
         }
 
+	    /**
+	     * Create and dispatch onMediaFileSelected Event
+	     *
+	     * @param {object}  data  The data for the detail
+	     *
+	     * @returns {void}
+	     */
+        function sendEvent(data) {
+	        const ev = new CustomEvent('onMediaFileSelected', {"bubbles":true, "cancelable":false, "detail": data});
+	        window.parent.document.dispatchEvent(ev);
+        }
         /**
          * Handle the click event
          * @param event
          */
         function handleClick(event) {
```

Comment:
We want to send the XHR when user Selects an supported file type and then when **he clicks the Select button**.
Not while he clicking every file on the Media Browser. Otherwise you will end up making your every file in a cloud being shareable which is a bad flaw